src/Fabulous.Avalonia/Widgets/Controls/Panels/ReversibleStackPanel.fs

```diff
@@ -0,0 +1,67 @@
+namespace Fabulous.Avalonia
+
+open Avalonia.Controls
+open Avalonia.Layout
+open Fabulous
+open System.Runtime.CompilerServices
+open Fabulous.StackAllocatedCollections
+
+type IFabReversibleStackPanel =
+    inherit IFabStackPanel
+
+module ReversibleStackPanel =
+    let WidgetKey = Widgets.register<ReversibleStackPanel> ()
+
+    let ReverseOrder =
+        Attributes.defineAvaloniaPropertyWithEquality ReversibleStackPanel.ReverseOrderProperty
+
+[<AutoOpen>]
+module ReversibleStackPanelBuilders =
+    type Fabulous.Avalonia.View with
+
+        static member VStack<'msg>(?spacing: float, ?reverseOrder: bool) =
+            let spacing =
+                match spacing with
+                | Some spacing -> spacing
+                | None -> 0.
```

Comment:
Optimization: When there is a `match ... with` condition on an optional argument in an inlined method, the F# compiler will resolve the condition at compile team and only keep the resulting value.

Say you write `VStack(spacing = 12.) { }`, the conditions will be removed at compile time and the code will look like:
```fs
let spacing = 12.
let reverseOrder = false

let scalers = (...)
```

We can take advantage of this optimization to improve even more the performance by avoiding to add the attribute to the widget altogether if the optional arguments are not passed.

This is possible by rewriting the code slightly differently:

```fs
static member inline VStack<'msg>(?spacing: float, ?reverseOrder: bool) =

    let mutable scalers = StackList.one(StackPanel.Orientation.WithValue(Orientation.Vertical))

    match spacing with
    | None -> ()
    | Some v -> scalers <- StackList.add(&scalers, StackPanel.Spacing.WithValue(spacing))

    match reverseOrder with
    | None -> ()
    | Some v -> scalers <- StackList.add(&scalers, ReversibleStackPanel.ReverseOrder.WithValue(reverseOrder))

    CollectionBuilder<'msg, IFabReversibleStackPanel, IFabControl>(
        ReversibleStackPanel.WidgetKey,
        scalers,
        Panel.Children
    )
```

So if you pass no argument, the code will be
```fs
    let scalers = StackList.one(StackPanel.Orientation.WithValue(Orientation.Vertical))

    CollectionBuilder<'msg, IFabReversibleStackPanel, IFabControl>(
        ReversibleStackPanel.WidgetKey,
        scalers,
        Panel.Children
    )
```

which is less work for Fabulous

Reply:
Thank you for taking the time to explaining that in detail! I've just committed the changes you recommended.